lite/kernels/nnadapter/converter/pad.cc

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) 2019 PaddlePaddle Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "lite/kernels/nnadapter/converter/converter.h"
+
+namespace paddle {
+namespace lite {
+namespace kernels {
+namespace nnadapter {
+
+int ConvertPad(Converter* converter, OpInfo* op, Scope* scope) {
+  // Extract op attributes
+  // Input
+  auto x_name = op->Input("X").front();
+  auto x_scale_name = "X0_scale";
+  std::vector<float> x_scales;
+  if (op->HasInputScale(x_scale_name, true)) {
+    x_scales = op->GetInputScale(x_scale_name, true);
+  }
+  // Mode
+  auto mode = op->GetAttr<std::string>("mode");
+  // Value, pads
+  float value;
+  std::vector<int> pads;
+  if (op->Type() == "pad2d") {
+    if (op->HasAttr("variable_padding") &&
+        op->GetAttr<bool>("variable_paddings")) {
```

Comment:
```suggestion
    CHECK (!HasInput(op, scope, "Paddings"));
```

PS：variable_paddings 是从 paddle 单测文件中迁移过来的错误用法，原定义中是没有这个参数的

Reply:
已修改